src/components/editor/TextArea.js

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import RIEStatefulBase from 'riek/lib/RIEStatefulBase';
+export default class TextArea extends RIEStatefulBase {
+    keyDown = (event) => {
+        if (event.keyCode === 27) { this.cancelEditing() }     // Escape
+    };
+
+    renderEditingComponent = () => {
+        return <textarea
+            rows={this.props.rows}
+            cols={this.props.cols}
+            disabled={this.state.loading}
+            className={this.makeClassString()}
+            defaultValue={this.props.value}
+            onInput={this.textChanged}
+            onBlur={this.finishEditing}
+            ref="input"
+            onKeyDown={this.keyDown}
+            {...this.props.editProps} />;
+    };
+
+    renderNormalComponent = () => {
+        const value = this.state.newValue || this.props.value
+        const spans_and_brs = []
+        var anySpan = false
+        let i = 0
+        value.split("\n").map(line => {
+            if(line.length > 0){
+                anySpan = true
+                spans_and_brs.push(<span key={i}>{line}</span>)
+            }
+          spans_and_brs.push(<br key={i+1} />)
+          i += 2
+        });
+
+        spans_and_brs.pop() // remove last br tag
+        if(!anySpan){
+            spans_and_brs.length=0
```

Comment:
I'm not sure you are supposed to ever just set the length of an array to 0 in javascript.  What is the purpose of this?  To clear it out?  If that is the case, just do `spans_and_brs = [];`  Or, in combination with the below line, do `spans_and_brs = [<span key="placeholder">{''}</span>]`;

Maybe I'm misreading the intent of this block though.

Reply:
Ah, but this would require you to change the `spans_and_brs` from `const` to `let` in the declaration.